editor/src/core/workers/linter/linter.ts

```diff
@@ -9,8 +9,17 @@ BrowserFS.configure({ fs: 'InMemory', options: {} }, (e) => {
     throw e
   }
 })
+if (typeof (globalThis as any).global === 'undefined') {
+  ;(globalThis as any).global = globalThis
+}
 ;(global as any).BrowserFS = BrowserFS
 
+// worker polyfills
+self.process = process ?? {}
+self.process.cwd = process.cwd ?? new Function()
+import { Buffer } from 'buffer'
+self.Buffer = Buffer
+
```

Comment:
Some (or all) of these shims need to happen elsewhere, but I'm not sure where :thinking: Perhaps in the entry points for each of the individual workers

Reply:
I have no extracted these to `vite-shims.ts`. The embedded preview still doesn't work when running with Vite though (Buffer is still not quite being shimmed correctly for the preview)